src/asset-picker/AssetPickerDropDown.js

```diff
@@ -35,26 +35,27 @@ export default class AssetPickerDropDown extends Component {
             router.push('asset-picker');
         }
     }
+    onClose(e) {
+        const { actions } = this.props;
+        e.stopPropagation();
+        actions.updateAssetPickerSearchQuery('');
+        actions.updateAssetPickerFilter('all');
```

Comment:
Add a new action instead called resetAssetPickerFilter.
We should keep double state access for exceptional cases, and not as a rule.